File: Backend/src/models/game-model.ts
import { Games } from "@prisma/client"

export interface CreateGameRequest {
    game_name: string
    income: number
    description: string
    currency_name: string
    tickets_name?: string
    image_url?: string
}

export interface GameResponse {
    game_id: number
    game_name: string
    income: number
    description: string
    currency_name: string
    tickets_name: string | null
    image_url: string | null
}

export function toGameResponse(game: Games): GameResponse {
    return {
        game_id: game.game_id,
        game_name: game.game_name,
        income: game.income,
        description: game.description,
        currency_name: game.currency_name,
        tickets_name: game.tickets_name,
        image_url: game.image_url
    }
}

File: Frontend/app/src/main/java/com/visprog/starchive/models/GameModel.kt

package com.visprog.starchive.models

import com.google.gson.annotations.SerializedName

data class GetGameResponse(
    val data: GameModel
)

data class GameModel(
    @SerializedName("game_id") val gameId: Int,
    @SerializedName("game_name") val gameName: String?,
    @SerializedName("income") val income: Float,
    @SerializedName("description") val description: String,
    @SerializedName("currency_name") val currencyName: String?,
    @SerializedName("tickets_name") val ticketsName: String?,
    @SerializedName("image_url") val imageUrl: String?,
    @SerializedName("budgets") val budgets: List<BudgetModel>?,
    @SerializedName("plans") val plans: List<PlanModel>?,
    @SerializedName("expenses") val expenses: List<ExpenseModel>?,
    @SerializedName("banners") val banners: List<BannerModel>?,
    @SerializedName("user_games") val userGames: List<UserGameModel>?,
    @SerializedName("hard_pities") val hardPities: List<HardPityModel>?,
    @SerializedName("articles") val articles: List<ArticleModel>?
)